addons/sourcemod/scripting/ttt/ttt_inventory.sp

```diff
@@ -0,0 +1,690 @@
+			}
+		}
+		
+		giveCredit *= randomFloat;
+		lootedCredits = RoundFloat(giveCredit);
+	}
+	
+	//Loot Items
+	randomFloat = GetRandomFloat();
+	char[][] lootedItems = new char[g_iItemLootMax][16];
+	int itemsLooted = 0;
+	ArrayList lootItemList = CreateArray(16);
+	if (randomFloat < chance[lootchance_item])
+	{
+		StringMapSnapshot items = invItems.Snapshot();
+		for (int i = 0; i < items.Length; i++)
+		{
+			char key[16];
+			items.GetKey(i, key, sizeof(key));
+			
+			int amount;
+			invItems.GetValue(key, amount);
+			
+			for (int j = 0; j < amount; j++)
+			{
+				lootItemList.PushString(key);
+			}
+		}
+		
+		float extraItemChance = chance[lootchance_item] * g_fLootChanceFallOff;
+		int maxLoots = g_iItemLootMax;
+		ArrayList itemIndex = CreateArray();
+		if (lootItemList.Length <= g_iItemLootMax)
+		{
+			maxLoots = lootItemList.Length;
+		}
+		
+		for (; itemsLooted < maxLoots; itemsLooted++) 
+		{
+			char itembuf[16];
+			int randomIndex = GetRandomInt(0, lootItemList.Length - 1);
+			if (itemIndex.FindValue(randomIndex) != -1)
+			{
+				itemsLooted--;
+				continue;
+			}
+			
+			lootItemList.GetString(randomIndex, itembuf, sizeof(itembuf));
+			strcopy(lootedItems[itemsLooted], 16, itembuf);
+			itemIndex.Push(randomIndex);
+			
+			randomFloat = GetRandomFloat();
+			if (randomFloat < extraItemChance)
+			{
+				extraItemChance *= g_fLootChanceFallOff;
+				continue;
+			}
+			
+			break;
+		}
+		
+	}
+	
+	if (lootedCredits > 0)
+	{
+		TTT_AddClientCredits(client, lootedCredits);
+	}
+	
+	for (int i = 0; i < itemsLooted; i++)
+	{
+		TTT_GiveClientItem(client, lootedItems[i]);
+	}
+	
+	CPrintToChat(client, "You looted %d credits and %d items from %s's corpse!", lootedCredits, itemsLooted, ragdoll[VictimName]);
+	return Plugin_Continue;
+}
+
+public int Native_AddInventoryItem(Handle plugin, int numparams)
+{
+	char itemshort[16];
+	int client = GetNativeCell(1);
+	int quantity = GetNativeCell(3);
+	GetNativeString(2, itemshort, sizeof(itemshort));
+	
+	if (quantity <= 0)
+	{
+		quantity = 1;
+	}
+	
+	return AddInventoryItem(client, itemshort, quantity);
+}
+
+public int Native_RemoveInventoryItem(Handle plugin, int numparams)
+{
+	char itemshort[16];
+	int client = GetNativeCell(1);
+	int amount = GetNativeCell(3);
+	GetNativeString(2, itemshort, sizeof(itemshort));
+	
+	return RemoveInventoryItem(client, itemshort, amount);
+}
+
+public int Native_GetInventoryListing(Handle plugin, int numparams)
+{
+	int client = GetNativeCell(1);
+	
+	StringMap inventoryListing = GetPlayerInventory(client);
+	
+	if (inventoryListing == null)
+	{
+		return view_as<int>(INVALID_HANDLE);
+	}
+	
+	return view_as<int>(inventoryListing.Snapshot());
+}
+
+public int Native_IsItemInInventory(Handle plugin, int numparams)
+{
+	char itemshort[16];
+	int client = GetNativeCell(1);
+	
+	GetNativeString(2, itemshort, sizeof(itemshort));
+	
+	return view_as<int>(IsItemInInventory(client, itemshort));
```

Comment:
why int? IsItemInInv... is just 0 or 1 so why not as bool?

Reply:
Hmm. 1AM Coding at it again.